src/bootstrap/lib.rs

```diff
@@ -65,9 +65,15 @@
 //! also check out the `src/bootstrap/README.md` file for more information.
 
 #![deny(warnings)]
+#![feature(associated_consts)]
+#![feature(core_intrinsics)]
```

Comment:
Ah, right. Well, I somewhat agree that it's not good that we need this; however, I haven't yet come up with a good way to avoid it -- @aidanhs I believe was going to look into changing the cache somehow to not need this.

The "straightforward" way to fix this is to generate a cache hashmap for each type with a macro. I don't really like the approach, but I *think* it'll work.